synapse/storage/data_stores/main/events_bg_updates.py

```diff
@@ -573,3 +577,63 @@ def _event_store_labels_txn(txn):
             yield self._end_background_update("event_store_labels")
 
         return num_rows
+
+    @defer.inlineCallbacks
+    def _event_store_expiry(self, progress, batch_size):
```

Comment:
we should only do this if ephemeral events are enabled.

Reply:
If we're going this way, then the background update sounds a bit useless to me. I'm deleting it (and we'll probably re-add it if we want to release ephemeral messages as a non-experimental feature)